sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Authentication/ContainerRegistryRefreshTokenCache.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+#nullable enable
+
+namespace Azure.Containers.ContainerRegistry
+{
+    internal class ContainerRegistryRefreshTokenCache
```

Comment:
It seems like there is a lot of repeated code (that is fairly intricate - at least to my somewhat uninformed eyes) between this class and [AccessTokenCache](https://github.com/Azure/azure-sdk-for-net/blob/fe1cf199136f3aea2af6a9a42bf7f42a7b2c9894/sdk/core/Azure.Core/src/Shared/BearerTokenChallengeAuthenticationPolicy.cs#L145). It might be helpful to include documentation on why this was necessary, or maybe even consider making AccessTokenCache sharable and perhaps derive from it here (if that would even make sense).

Reply:
I can add comments to this effect.  This is a stop-gap measure, and it sounds like it's probably not a great time to rewrite AccessTokenCache.  I'm changing it as little as possible.